### PR TITLE
r/iam_saml_provider - refactor to use arn package + disappearing test case

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSIAMSamlProvider_'
--- PASS: TestAccAWSIAMSamlProvider_basic (70.74s)
--- PASS: TestAccAWSIAMSamlProvider_disappears (29.05s)
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSIAMSamlProvider_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSIAMSamlProvider_ -timeout 120m
=== RUN   TestAccAWSIAMSamlProvider_basic
=== PAUSE TestAccAWSIAMSamlProvider_basic
=== RUN   TestAccAWSIAMSamlProvider_disappears
=== PAUSE TestAccAWSIAMSamlProvider_disappears
=== CONT  TestAccAWSIAMSamlProvider_basic
=== CONT  TestAccAWSIAMSamlProvider_disappears
--- PASS: TestAccAWSIAMSamlProvider_disappears (16.54s)
--- PASS: TestAccAWSIAMSamlProvider_basic (35.16s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	35.203s
```

File: aws/resource_aws_iam_saml_provider.go [hunk 1]
```diff
@@ -108,22 +107,23 @@ func resourceAwsIamSamlProviderUpdate(d *schema.ResourceData, meta interface{})
...
-func extractNameFromIAMSamlProviderArn(arn, partition string) (string, error) {
-	// arn:aws:iam::123456789012:saml-provider/tf-salesforce-test
-	r := regexp.MustCompile(fmt.Sprintf("^arn:%s:iam::[0-9]{12}:saml-provider/(.+)$", partition))
-	submatches := r.FindStringSubmatch(arn)
-	if len(submatches) != 2 {
-		return "", fmt.Errorf("Unable to extract name from a given ARN: %q", arn)
+func extractNameFromIAMSamlProviderArn(samlArn string) (string, error) {
+	parsedArn, err := arn.Parse(samlArn)
+	if err != nil {
+		return "", fmt.Errorf("Unable to extract name from a given ARN: %q", parsedArn)
```
Comment (anGie44) [inline aws/resource_aws_iam_saml_provider.go:123]: Nit: in the event of error, i think passing the `samlArn` here in the error message will be helpful as `parsedArn` will be empty

Comment (DrFaust92) [reply aws/resource_aws_iam_saml_provider.go:123]: Missed that, will fix

Comment (DrFaust92) [reply aws/resource_aws_iam_saml_provider.go]: fixed

**Pull request merged**